### PR TITLE
feat: switch to physics v2.0 and add data validity range tests

### PR DESCRIPTION
**Pull request by kyle-cochran**
Updates to OSTk Physics `v2.0`. Add some lightweight tests to the `Propagator.test.cpp` to try and generate states at the edge of where input data is available. Those limits are:
- 5 years in the past when propagating with drag
- 1 year in the future when doing ITRF/TEME frame conversions

(these don't cover all propagation cases obviously, but do flex the data handling which is the intent 😁).

TODO: this won't even build in CI until Physics 2.0 is released, but was cross-tested locally.

### CODE REVIEW

File: test/OpenSpaceToolkit/Astrodynamics/NumericalSolver.test.cpp [hunk 1]
```diff
@@ -833,8 +833,6 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_NumericalSolver, IntegrateTime_Conjunctive
         // Ensure that integration terminates at condition if condition is met
         EXPECT_TRUE(propagatedTime != endTime);
         EXPECT_TRUE(conditionSolution.conditionIsSatisfied);
-        std::cout.precision(16);
```
Comment (vishwa2710) [inline test/OpenSpaceToolkit/Astrodynamics/NumericalSolver.test.cpp:836]: Nice thanks man.

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp [hunk 2]
```diff
@@ -2152,3 +2156,151 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator, FromEn
...
+
+TEST_P(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator_Data_Success, EdgeCases_Input_Data_Horizon)
+{
+    // long-horizon, low granularity propagation to test input data availability
+
+    // currently limited by:
+    // -5 years in the past due to CSSI space weather data when computing drag
+    // +1 year in the future due to finals2000a prediction span when computing polar motion for frame conversions
+
+    auto parameters = GetParam();
```
Comment (vishwa2710) [inline test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp:2193]:
```suggestion
    const auto parameters = GetParam();
```
Nit:

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp [hunk 3]
```diff
@@ -2152,3 +2156,151 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator, FromEn
...
+
+    Instant startInstant = std::get<0>(parameters);
+    Instant endInstant = std::get<1>(parameters);
+    Duration step = std::get<2>(parameters);
```
Comment (vishwa2710) [inline test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp:2197]:
```suggestion
    const Instant startInstant = std::get<0>(parameters);
    const Instant endInstant = std::get<1>(parameters);
    const Duration step = std::get<2>(parameters);
```
God why couldn't const be the default. #Rust

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp [hunk 4]
```diff
@@ -2152,3 +2156,151 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator, FromEn
...
+
+    const Shared<Celestial> earthSPtr = std::make_shared<Celestial>(earth);
+    const Array<Shared<Dynamics>> dynamics = {
+        std::make_shared<PositionDerivative>(),
+        std::make_shared<CentralBodyGravity>(earthSPtr),
+        std::make_shared<AtmosphericDrag>(earthSPtr, satelliteSystem_),
+    };
+
+    // Setup Propagator model and orbit
+    const Propagator propagator = {RK4, dynamics};
```
Comment (vishwa2710) [inline test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp:2228]:
```suggestion
    const Propagator propagator = {
        RK4,
        dynamics,
    };
```

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp [hunk 5]
```diff
@@ -2152,3 +2156,151 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator, FromEn
...
+            Instant::Now() - Duration::Weeks(52 * 5),
+            Instant::Now() - Duration::Weeks(52 * 4.9),
+            Duration::Days(7),
+            "NRLMSISE Past Data limit pass"
+        ),
+        std::make_tuple(
+            Instant::Now() + Duration::Weeks(52 * 0.9),
+            Instant::Now() + Duration::Weeks(52 * 1.0),
+            Duration::Days(7),
+            "IERS Finals2000A Future Data limit pass"
```
Comment (vishwa2710) [inline test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp:2180]: Is this parameter being used?

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp [hunk 6]
```diff
@@ -2152,3 +2156,151 @@ TEST_F(OpenSpaceToolkit_Astrodynamics_Trajectory_Orbit_Models_Propagator, FromEn
...
+            Instant::Now() - Duration::Weeks(52 * 6.0),
+            Instant::Now() - Duration::Weeks(52 * 5.9),
+            Duration::Days(7),
+            "NRLMSISE Past Data limit fail"
+        ),
+        std::make_tuple(
+            Instant::Now() + Duration::Weeks(52 * 1.0),
+            Instant::Now() + Duration::Weeks(52 * 1.1),
+            Duration::Days(7),
+            "IERS Finals2000A Future Data limit fail"
```
Comment (vishwa2710) [inline test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp:2254]: Same. is this being used?

**Review by vishwa2710 — APPROVED** [review]
Looks good man! Just some nits.

File: test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp (hunk 5 above)
Comment (kyle-cochran) [reply test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp:2180]: It's not, but I thought it was useful to describe the purpose of the test case. If it grosses you out, I can make it a comment 😄

Comment (vishwa2710) [reply test/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.test.cpp:2180]: Fair point. So you can actually make it print out a name for the test like so:
```
INSTANTIATE_TEST_SUITE_P(
    MyInstantiation, MyTestSuite,
    testing::Values(...),
    [](const testing::TestParamInfo<MyTestSuite::ParamType>& info) {
      return info.param[3];
    });
```
Something like that.

**Pull request merged**